lib/mage/Mage.js

```diff
@@ -91,6 +107,30 @@ function Mage(mageRootModule) {
 
 util.inherits(Mage, EventEmitter);
 
+/**
+ * If called, this should be called from within you project's
+ * `./lib/index` file, or equivalent.
+ *
+ * It is also noteworthy that in the case of TypeScript, if you
+ * choose to manually transpile your project to JavaScript, you
+ * should then set the project type to JavaScript instead; if you
+ * are using 'ts-node' (this is default for bootstrapped MAGE TypeScript
+ * projects), then the type should be TypeScript.
+ *
+ * @summary Set the type of project (JavaScript or TypeScript)
+ * @param {string} projectType Type of project ('javascript' or 'typescript')
+ * @returns {Mage} This MAGE instance
+ * @throws {Error} Thrown if the project type is unknown
+ */
+Mage.prototype.setProjectType = function (projectType) {
```

Comment:
I'm not too thrilled about the name of this method. Not sure I can give any better alternatives though, but a "project type" could be anything (eg: "game").